runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -77,19 +77,33 @@ impl TrieViewer {
         state_update: &TrieUpdate,
         account_id: &AccountId,
         public_key: &PublicKey,
-    ) -> Result<AccessKey, errors::ViewAccessKeyError> {
-        get_access_key(state_update, account_id, public_key)?.ok_or_else(|| {
-            errors::ViewAccessKeyError::AccessKeyDoesNotExist { public_key: public_key.clone() }
-        })
+        include_proof: bool,
+    ) -> Result<ViewAccessKeyResult, errors::ViewAccessKeyError> {
+        let prefix = trie_key_parsers::get_raw_prefix_for_access_key(account_id, public_key);
+
+        let mut iter = state_update.trie().iter()?;
+        iter.remember_visited_nodes(include_proof);
+        iter.seek_prefix(&prefix)?;
+
+        let access_key =
+            get_access_key(state_update, account_id, public_key)?.ok_or_else(|| {
+                errors::ViewAccessKeyError::AccessKeyDoesNotExist { public_key: public_key.clone() }
+            })?;
+
+        let proof = iter.into_visited_nodes();
+
+        Ok(ViewAccessKeyResult { access_key, proof })
     }
 
+    // TODO - Add boolean proof parameter support
```

Comment:
if you're adding a TODO - please create a github issue and mention its number here.

Reply:
[e2cded2](https://github.com/near/nearcore/pull/8644/commits/e2cded2a2c205f5cf907dd1e22d0ca2833a6a63f)